test/helper.js

```diff
@@ -58,6 +54,51 @@ class FakeRoles {
   }
 }
 
+class FakePublisher {
```

Comment:
For this to be a generic pattern that can be repeated in other services,  the `FakePublisher` class has to be derived from the `Exchanges` definition in `exchanges.js`.

This means tweaking the `pulse-publisher` package so that creation of such a class is supported. In effect we should have `const FakePublisher = require('../src/exchanges').createFakePublisher()`